servicetalk-http-api/src/main/java/io/servicetalk/http/api/HeaderUtils.java

```diff
@@ -675,6 +675,13 @@ static boolean hasContentType(final HttpHeaders headers,
             return true;
         }
 
+        if (UTF_8.equals(expectedCharset) &&
```

Comment:
@sonicloong we are doing this and other checks above too, can we avoid doing multiple checks here, eg: by storing prior results in a local variable?

Reply:
@NiteshKant that makes sense. I've consolidated the duplicate `UTF_8.equals`/`contentEqualsIgnoreCase` checks.